zulip_bots/zulip_bots/lib.py

```diff
@@ -218,6 +246,16 @@ def handle_message(message):
                 return
 
         if is_private_message or is_mentioned:
+            # Handle any default commands first
+            if len(updated_defaults) > 0:
+                for command in updated_defaults:
+                    if command == '' and len(message['content']) == 0:
+                        restricted_client.send_reply(message, updated_defaults[''][0]())
+                        return
+                    if command == message['content']:
+                        restricted_client.send_reply(message, updated_defaults[command][0]())
```

Comment:
To avoid accessing our data structures with `[0]` (which makes code harder to understand), I suggest to implement command_defaults as a dict of dicts, which could look like this:

```python
command_defaults = {
         '': {'function': default_empty_response, 'explanation': "[BLANK MESSAGE NOT SHOWN]"},
         'about': {'function': default_about_response, 'explanation': "The type and use of this bot"},
         'usage': 'function': {lambda: message_handler.usage(), 'explanation': "Bot-provided usage text"},
     }
```

With nicer line-breaks than here oc :)

This would also allow us to not loop over all commands, but to directly access a command with a key.

Reply:
Implemented this way for now, though I would also consider a named-tuple approach.